Back-end/tests/com/smartplanner/BruteForcerTest.java

```diff
@@ -0,0 +1,126 @@
+package com.smartplanner;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import static org.junit.jupiter.api.Assertions.*;
+
+class BruteForcerTest{
+    ArrayList<Activity> activities;
+    BruteForcer bruteForcer;
+    int allPossibilities;
+
+    @BeforeEach
+    void beforeEach() {
+        activities = new ArrayList<Activity>();
+
+        ArrayList<Term> termsAct1 = new ArrayList<Term>();
+        termsAct1.add(new Term(10, 1, LocalTime.of(12, 30)));
+        termsAct1.add(new Term(20, 2, LocalTime.of(12, 30)));
+        activities.add(new Activity("activity1", termsAct1));
+
+        ArrayList<Term> termsAct2 = new ArrayList<Term>();
+        termsAct2.add(new Term(30, 3, LocalTime.of(12, 30)));
+        termsAct2.add(new Term(40, 4, LocalTime.of(12, 30)));
+        activities.add(new Activity("activity2", termsAct2));
+
+        ArrayList<Term> termsAct3 = new ArrayList<Term>();
+        termsAct3.add(new Term(50, 5, LocalTime.of(12, 30)));
+        termsAct3.add(new Term(60, 6, LocalTime.of(12, 30)));
+        activities.add(new Activity("activity3", termsAct3));
+
+        bruteForcer = new BruteForcer(activities, 7);
+        allPossibilities=termsAct1.size()*termsAct2.size()*termsAct3.size();
+    }
+
+    @Test
+    void isNextReturnsTrueIfThereIsNextCombinationToGet() {
+        for(int i=0; i<allPossibilities; ++i, bruteForcer.getNext())
+            assertTrue(bruteForcer.isNext());
+
+        assertFalse(bruteForcer.isNext());
+    }
+
+    @Test
```

Comment:
this test is huuuuuuuge. But maybe there is no better way to do it. Think about it if you have some time.